libexec/rtld-elf/aarch64/rtld_c18n_asm.S

```diff
@@ -334,6 +334,20 @@ PATCH_POINT(tramp_save_caller, cookie, 1b)
 PATCH_POINT(tramp_save_caller, ret_args, 2b)
 PATCH_POINT(tramp_save_caller, target, 3b)
 
+/*
+ * Save the address of the current frame to c29 so that unwinders can locate it.
+ * When transitioning to Restricted mode code, its tag must be cleared.
+ */
+TRAMP(tramp_update_fp_exe)
+	mov	c29, TRUSTED_STACK
+TRAMPEND(tramp_update_fp_exe)
+
+TRAMP(tramp_update_fp_res)
+	mov	x29, TRUSTED_STACK_X
+TRAMPEND(tramp_update_fp_res)
+#undef	TRUSTED_STACK
+#undef	TRUSTED_STACK_X
```

Comment:
I don't love the macros being defined within tramp_save_caller, leaking out past it and being used here

Reply:
I cannot think of a better way. Redefining them seems error prone. Moving the `#define`s before the first fragment might cause erroneous usage of `TRUSTED_STACK` before the register has actually been written to.

Reply:
That's trivial to audit, and nothing in this file stops the different fragments being combined in an order where that's still the case. That's the whole point, really; either you have it defined within a function, or you have it defined globally. Mixing the two just ends up with pretending you're enforcing something that you're not really doing, and gets confusing.